test.py

```diff
@@ -28,6 +28,34 @@ def test_deparam(self):
             good = base + good
             self.assertEqual(url.parse(bad).deparam(['c']).utf8(), good)
 
+    def test_deparam_case_insensitivity(self):
+        '''Parameters to remove can be supplied in any case.'''
+        examples = [
+            ('?hELLo=2', ''),
+            ('?HELLo=2', '')
+        ]
+        base = 'http://testing.com/page'
+        for bad, good in examples:
+            bad = base + bad
+            good = base + good
+            self.assertEqual(url.parse(bad).deparam(['HeLlO']).utf8(), good)
```

Comment:
I know this has been the pattern already used, but I would prefer there to be one assertion per test and use some metaprogramming to generate the separate test functions if you need multiple cases similar to [this answer](http://stackoverflow.com/a/3772008/1307166).